File: app/src/main/java/ru/geekbrains/notes/data/CardSource.java
package ru.geekbrains.notes.data;

public interface CardSource {
    int size();
    CardData getCardData(int position);
    CardSource init(CardsSourceResponse cardsSourceResponse);

    void deleteCardData(int position);
    void updateCardData(int position, CardData newCardData);
    void addCardData(CardData newCardData);
    void clearCardData();
}

File: app/src/main/java/ru/geekbrains/notes/data/CardSource.kt
package ru.geekbrains.notes.data

interface CardSource {
    fun size(): Int
    fun getCardData(position: Int): CardData
    fun init(cardsSourceResponse: CardsSourceResponse?): CardSource?
    fun deleteCardData(position: Int)
    fun updateCardData(position: Int, newCardData: CardData)
    fun addCardData(newCardData: CardData)
    fun clearCardData()
}
